### PR TITLE
[flutter_plugin_tools] Validate code blocks in readme-check

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates `readme-check` to:
- Require that a language annotation to be added to any code block.
- Optionally require that any Dart (for now; we could eventually extend to other
  languages) code block use the `code-excerpt`.

Adds CI to validate that all packages are using `code-excerpt`, with a
temporary opt-out list for all of the packages with existing code blocks,
so that we can adopt it incrementally but prevent any new violations.

In order to land the language tag requirement, fixes a couple of blocks
that were missing the annotation.

No version change: The addition of a language block tag is too minor to be worth releasing.
No CHANGELOG change: The addition of a language block tag is too minor to be worth noting.

Part of https://github.com/flutter/flutter/issues/102679

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/tool/test/readme_check_command_test.dart [hunk 1]
```diff
@@ -275,4 +275,135 @@ A very useful plugin.
...
+    test('passes when excerpt requirement is met', () async {
+      final Directory packageDir = createFakePackage('a_package', packagesDir);
+
+      packageDir.childFile('README.md').writeAsStringSync('''
+Example:
+
+<?code-excerpt "main.dart (SomeSection)"?>
+```dart
+A B C
+```
```
Comment (ditman) [inline script/tool/test/readme_check_command_test.dart:366]: Is this what the output of the file will look like after running code-excerpt? Adding the code block below the code-excerpt tag?

**Review by ditman — APPROVED** [review]
I'm kinda excited about this. Not thrilled about the submodule, I wish the code-excerpter thing was a more proper "dependency".

File: script/tool/test/readme_check_command_test.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/test/readme_check_command_test.dart:366]: Yep, you put it above a code block (which can be empty to start) and the script will fill it with the extracted code. You can take a look at the camera README for a real example.

**Comment by stuartmorgan** [comment]
> Not thrilled about the submodule

I hear you; my last experience with submodules (which granted was a while ago) was decidedly not great. But since this is firewalled off to just this script, I'm hoping it'll be pretty painless.

File: script/tool/test/readme_check_command_test.dart (hunk 1 above)
Comment (ditman) [reply script/tool/test/readme_check_command_test.dart:366]: I think I've seen it in the Flutter docs site, and it was not super evident how to update the `.md` snippets (or the link to actual code somewhere else).

I think I ended up doing a find and replace first, then learning about a script being available somewhere, and *eventually* managing to run it a while later; here are David's Misadventures With Code-Excerpt As A Complete Noob:

* https://github.com/flutter/website/pull/6933

Comment (stuartmorgan) [reply script/tool/test/readme_check_command_test.dart:366]: I added https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#readme-code

Hopefully it's reasonably clear; feedback welcome!

**Pull request merged**